data/src/main/kotlin/com/vultisig/wallet/data/repositories/TokenRepository.kt

```diff
@@ -139,6 +139,10 @@ internal class TokenRepositoryImpl @Inject constructor(
 
     override suspend fun getTokenByContract(chainId: String, contractAddress: String): Coin? {
         val chain = Chain.fromRaw(chainId)
+        if (chain == Chain.Solana)
+            return splTokenRepository.getTokenByContract(
+                contractAddress
+            ) ?: return null
```

Comment:
Hi @aminsato , this will not really scale when supporting other chains. Lets fix the root cause and work with better abstractions

I suggest :
- Renaming getTokenByContract -> getEVMTokenByContract
- Create a generic getTokenByContract class that call specific repository